jina/types/document/__init__.py

```diff
@@ -287,6 +287,7 @@ def _update_doc(d: Dict):
             )
         self.set_attributes(**kwargs)
         self._mermaid_id = random_identity()  #: for mermaid visualize id
+        self.update_content_hash()
```

Comment:
This is very dangerous, why if the `content_hash` is already computed and properly stored in the `proto`? Then u do not want to update

Reply:
Handled by using a flag, that is set to False at iteration